File: AdamClassLib/Sample Code/APAX/WinCE/VB/APAX-Coupler-Sample/Apax_IO_Module_Library/Wait_Form.vb
Public Class Wait_Form
    Dim iCount As Integer = 1
    Dim iRunTime As Integer = 0
    Dim iMilliSec As Integer = 0
    Public Sub Start_Wait()

        Me.iMilliSec = 4500
        Me.Timer1.Enabled = True

    End Sub
    Public Sub Start_Wait(ByVal iWait As Integer)

        Me.iMilliSec = iWait
        Me.Timer1.Enabled = True

    End Sub
    Public Sub End_Wait()

        Me.Timer1.Enabled = False
        Me.Close()


    End Sub
    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        Dim iLabelStatus As Integer

        iLabelStatus = (iCount Mod 4)
        iRunTime = iRunTime + Timer1.Interval

        Select Case (iLabelStatus)
            Case 0
                Me.lbl_Wait.Text = "Please waiting."
                iCount = 0
            Case 1
                Me.lbl_Wait.Text = "Please waiting.."
            Case 2
                Me.lbl_Wait.Text = "Please waiting..."
            Case 3
                Me.lbl_Wait.Text = "Please waiting...."
        End Select

        If (iRunTime >= iMilliSec) Then

            progressBar1.Value = 100
            End_Wait()

        Else

            progressBar1.Value = iRunTime * 100 / iMilliSec

        End If
        iCount = iCount + 1
    End Sub

End Class

File: AdamClassLib/Sample Code/APAX/Win32/CSharp/APAX-PAC-Sample/APAX_IO_Module_Library/Wait_Form.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Apax_IO_Module_Library
{
    public partial class Wait_Form : Form
    {
        private int iCount;
        private int iRunTime;
        private int iMilliSec;

        public Wait_Form()
        {
            InitializeComponent();
            iCount = 1;
            iRunTime = 0;
            iMilliSec = 0;

        }
        public void Start_Wait()
        {

            this.iMilliSec = 4500;
            this.Timer1.Enabled = true;
        }
        public void Start_Wait(int iWait)
        {
            this.iMilliSec = iWait;
            this.Timer1.Enabled = true;
        }
        public void End_Wait()
        {
            Timer1.Enabled = false;
            this.Close();
        }
        private void Timer1_Tick(object sender, EventArgs e)
        {
            int iLabelStatus;
            iLabelStatus = (iCount % 4);
            iRunTime = (iRunTime + Timer1.Interval);

            switch (iLabelStatus)
            {
                case 0:
                    this.lbl_Wait.Text = "Please waiting.";
                    iCount = 0;
                    break;
                case 1:
                    this.lbl_Wait.Text = "Please waiting..";
                    break;
                case 2:
                    this.lbl_Wait.Text = "Please waiting...";
                    break;
                case 3:
                    this.lbl_Wait.Text = "Please waiting....";
                    break;
            }
            if ((iRunTime >= iMilliSec))
            {
                progressBar1.Value = 100;
                End_Wait();
            }
            else
            {
                progressBar1.Value = (int)(iRunTime * (100.0 / (double)iMilliSec));
            }
            iCount = (iCount + 1);
        }

    }
}